lib/assign.js

```diff
@@ -32,16 +32,297 @@ mysql> select action,count(*) from mdl_log where module='assign' group by action
 */
 
 var library = {
-    "submit": undefined
+    "add":{
         /*
-        | userid | course | cmid | url              | info              |
-        +--------+--------+------+------------------+-------------------+
-        |   1296 |    103 | 9757 | view.php?id=9757 | <html text>       |
+         mysql> SELECT * FROM mdl_log WHERE module='assign' AND action = 'add' ORDER BY id DESC LIMIT 1;
+        +---------+------------+--------+----------------+--------+--------+-------+--------+-------------------+------+
+        | id      | time       | userid | ip             | course | module | cmid  | action | url               | info |
+        +---------+------------+--------+----------------+--------+--------+-------+--------+-------------------+------+
+        | 1976864 | 1424085174 |    185 | 10.111.112.125 |    274 | assign | 24348 | add    | view.php?id=24348 | 3937 |
+        +---------+------------+--------+----------------+--------+--------+-------+--------+-------------------+------+
 
-        cmid doesn't relate to mdl_assign.id or mdl_assign_submission.id (in our dataset?).
-        Need to read some code...
+        userid  => mdl_user.id
+        course  => mdl_course.id
+        module  => mdl_module.name
+        cmid    => mdl_course_modules.id
+        url: id => cmid
+        info    => mdl_assign.id
+
+        mysql> SELECT count(*) FROM mdl_log WHERE module='assign' AND action = 'add';
+        +----------+
+        | count(*) |
+        +----------+
+        |      315 |
+        +----------+
         */
+
+        sql_old:
+            'SELECT log.*, ' +
+                'u.username AS u_username, u.email AS u_email, ' +
+                'c.shortname AS course_shortname, ' +
+                'a.name AS assign_name ' +
+                'FROM mdl_log log ' +
+                'LEFT JOIN mdl_course c ON log.course = c.id ' +
+                'LEFT JOIN mdl_course_modules cm ON log.cmid = cm.id ' +
+                'LEFT JOIN mdl_assign a ON cm.instance = a.id ' +
+                'JOIN mdl_user u ON log.userid = u.id ' +
+                "WHERE log.module = 'assign'" +
+                "AND log.action = 'add'" +
+                "AND " + restrict_clause,
+
+        sql_old_2pass: (row) => {
+            var m = row.url.match(/id=(\d+)/);
+            if (!m){
+                console.log('No course module ID in URL ' + JSON.stringify(row));
+                return null;
+            }
+        },
+
+        sql_match: (row) => {
+            return row.cm_id ?
```

Comment:
There's no such field in the result of your sql_old query. Do you mean row.cmid?